src/main/java/org/traccar/protocol/Gt06ProtocolDecoder.java

```diff
@@ -794,22 +794,33 @@ private Object decodeBasic(Channel channel, SocketAddress remoteAddress, ByteBuf
 
             position.set(Position.KEY_TYPE, type);
 
+            if ("NT20".equals(model)) {
+                buf.readUnsignedByte(); // location source type
+                buf.skipBytes(8); // imei
+
+                DateBuilder dateBuilder = new DateBuilder((TimeZone) deviceSession.get(DeviceSession.KEY_TIMEZONE))
+                        .setDate(buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte())
+                        .setTime(buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte());
+
+                position.setDeviceTime(dateBuilder.getDate());
+            }
```

Comment:
Why do we need this instead of just using `decodeGps`?

Reply:
it's not instead, decodeGps is called after that.

Reply:
that's not related to decodeGps, we call that after

Reply:
I see. There are two timestamps. But it looks like it only applies to message type 0x22.

Reply:
That message is used in other models, apparently with a different structure. The tests will fail if we check the message type instead of the model.

Reply:
Yes, I mean that we need to check both. Make sure your decoding logic is valid for the whole protocol.